languages/csharp/exercises/concept/properties/.meta/design.md

```diff
@@ -0,0 +1,193 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to introduce the student to C# properties and to teach the student how the concept of properties is implemented in [C#][docs.microsoft.com-properties]. We'll teach the student about properties by having the student work with classes that expose contrasting forms of properties. The students will learn to work with properties with backing fields and auto-implemented properties.
+
+Properties are covered early in the C# track as their purpose and power can be shown with few dependencies (classes, access modifiers and fields of simple types).
+
+
+## Things to teach
+
+- public properties
+- properties with backing fields
+- auto-implemented properties 
+- input verification
+- calculated properties
+- get-only properties vs. properties with private set accessors
+- set-only properties vs. properties with private get accessors
+- properties exposing reference types
+
+There is no way to enforce the use of backing fields in preference to auto-implemented properties
+or vice versa either through tests or by giving a steer in the scenario.  The best we can hope for
+is that mentors' comments offer some opinion and the alternative.
+
+Similarly there is no way to distinguish between set/get-only properties and those with
```

Comment:
I think we actually _can_ do this through [reflection](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo?view=netframework-4.8).

Reply:
Sorry to say I had not considered the challenge for implementors of analysers and representers!  It was more about whether we can steer the student so that they become aware of and exercise various different constructs.  It is useful in mentoring if there is one clear favourite solution (particularly as we are concerned more with correct language usage than software craftsmanship).  When it came to implementing Example.cs this seemed to resolve itself better than I expected.  I will change the wording to reflect my perspective.

Reply:
Great! And steering the student towards a specific solution is hard.